colossus/src/main/scala/colossus/service/ServiceDSL.scala

```diff
@@ -185,9 +151,30 @@ trait ClientLifter[C <: Protocol, T[M[_]] <: Sender[C,M]] {
 trait ClientFactory[C <: Protocol, M[_], T <: Sender[C,M], E] {
 
 
+  /**
+    * Load a ServiceClient definition from a Config.  Looks into `colossus.clients.$clientName` and falls back onto
+    * `colossus.client-defaults`
+    * @param clientName The name of the client definition to load
+    * @param config A config object which contains at the least a `colossus.clients.$clientName` and a `colossus.client-defaults`
+    * @return
+    */
+  def apply(clientName : String, config : Config = ConfigFactory.load())(implicit provider: ClientCodecProvider[C], env: E) : T = {
+    apply(ClientConfig.load(clientName, config))
+  }
+
+  /**
+    * Create a Client from a config source.
+    *
+    * @param config A Config object in the shape of `colossus.client-defaults`.  It is also expected to have the `address` and `name` fields.
+    * @return
+    */
+  def apply(config : Config)(implicit provider: ClientCodecProvider[C], env: E) : T = {
+    apply(ClientConfig.load(config))
+  }
+
   def apply(config: ClientConfig)(implicit provider: ClientCodecProvider[C], env: E): T
 
-  def apply(host: String, port: Int, requestTimeout: Duration = 1.second)(implicit provider: ClientCodecProvider[C], env: E): T = {
+  def apply(host: String, port: Int, requestTimeout: Duration)(implicit provider: ClientCodecProvider[C], env: E): T = {
```

Comment:
I think I'd be more in favor of just dropping the `requestTimeout` param instead of making it mandatory.  I think in most cases if you're going to be configuring a timeout you're gonna want to pass in a whole config and set a name as well.  This constructor is really just for demos and hello-world-style stuff.